compiler/lib/src/main/scala/codegen/CppWriter/CppWriter.scala

```diff
@@ -25,7 +25,12 @@ object CppWriter extends AstStateVisitor with LineUtils {
   ) = {
     val node = aNode._2
     val data = node.data
-    visitList(s, data.members, matchComponentMember)
+    val cppDoc = ComponentCppWriter(s, aNode).write
+    for {
+      _ <- writeCppDoc(s, cppDoc)
```

Comment:
I think the LHS should be s in line 30 (we should thread the state through the computation). Even if the state isn't changed in line 30, this is good defensive programming.